src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1166,6 +1171,49 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             return expression;
         }
 
+        /// <summary>
+        /// Is this is an _ identifier in a context where discards are allowed?
+        /// </summary>
+        private static bool FallBackOnDiscard(IdentifierNameSyntax node, DiagnosticBag diagnostics)
+        {
+            if (node.Identifier.ContextualKind() != SyntaxKind.UnderscoreToken)
+            {
+                return false;
+            }
+
+            CSharpSyntaxNode containingDeconstruction = node.GetContainingDeconstruction();
+            return containingDeconstruction != null || IsOutVarDiscardIdentifier(node);
+        }
+
+        private static bool IsOutVarDiscardIdentifier(SimpleNameSyntax node)
+        {
+            Debug.Assert(node.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken);
+
+            CSharpSyntaxNode parent = node.Parent;
+            if (parent?.Kind() != SyntaxKind.Argument ||
+                ((ArgumentSyntax)parent).RefOrOutKeyword.Kind() != SyntaxKind.OutKeyword)
```

Comment:
If any out argument should be allowed, it feels like there is no reason to do the extra checks.